leader/leader.go

```diff
@@ -71,7 +60,7 @@ func Become(ctx context.Context, lockName string) error {
 
 	ns, err := getOperatorNamespace()
 	if err != nil {
-		if err == errNoNamespace || err == errRunLocal {
+		if err == errNoNamespace {
 			log.Info("Skipping leader election; not running in a cluster.")
 			return nil
 		}
```

Comment:
I think if I was using this leader election library, I would want the ability to handle the no namespace error myself. WDYT of exporting `ErrNoNamespace` so that callers can do:
```go
if err := leader.Become(...); err != nil {
	if errors.Is(err, leader.ErrNoNamespace) {
		// handle the no namespace error
	}
	// other error handling
}
```

And then just having this check return the error directly.